pkg/specgen/generate/kube/volume.go

```diff
@@ -271,6 +271,10 @@ func InitializeVolumes(specVolumes []v1.Volume, configMaps []v1.ConfigMap, secre
 			return nil, fmt.Errorf("failed to create volume %q: %w", specVolume.Name, err)
 		}
 
+		// set the volume name to be exactly
+		// the same as specified in the k8s yaml
+		volume.Source = specVolume.Name
```

Comment:
This cannot be overwritten here, it must be set in the proper `VolumeFrom...` functions, i.e. right now this overwrites the host path.

Reply:
Right, sorry about that. Made some changes to `VolumeFromConfigMap`, I'll check what's the behavior with other `VolumeFrom...` functions.

Reply:
Quoting from another comment of mine:

> But in the longer run, I believe KubeVolume could have a Name attribute which would be the direct equivalent of the volume name in the k8s manifest. That is of course, I'm missing out on a specific design decision that involved using Source as the volume name for all kinds of volumes (the current implementation).